pkg/manifestclient/list.go

```diff
@@ -74,6 +75,10 @@ func (mrt *manifestRoundTripper) listAll(requestInfo *apirequest.RequestInfo) ([
 		if retList.GroupVersionKind() != kind.listKind {
 			return nil, fmt.Errorf("inconsistent list kind: got %v, expected %v", retList.GroupVersionKind(), kind.listKind)
 		}
+		retList, err := filterByLabelSelector(retList, requestInfo.LabelSelector)
+		if err != nil {
+			return nil, fmt.Errorf("failed to filter by labelSelector %s: %w", requestInfo.LabelSelector, err)
```

Comment:
nit: the error msg already contains the labelSelector.

maybe we should just return the received errr:
```
return nil, err
```